### DISCUSSION TITLE
Monitoring tool for latency, connection errors & throughput spikes

### DISCUSSION THREAD

**Discussion by jnmayur** [processone/ejabberd#4063] [Questions] [answered]
Hello folks

I'm trying to figure out if there's an open source monitoring tool available for ejabberd. We're using a cluster of two nodes but we don't have any monitoring other than basic instance level CPU, memory & page faults tracking.

If latencies in sending messages & connecting to ejabberd can be tracked that would be of great help.

**Comment by mremond** [comment]
ejabberd does not support this latency tracking. Latency monitoring is probably more accurately tracked outside ejabberd, by using a custom client to monitor and report to a graphing tool.

**Reply by rhariady** [reply]
it may not directly related, but is there any way to monitor number of messages being send/received/process at the moment?

**Reply by badlop** [reply, marked as answer]
I don't know if they provide exactly what you want, but there are two contributed modules that print statistics in different formats:
- https://github.com/lemon2015/ejabberd-prometheus-exporter
- https://github.com/processone/ejabberd-contrib/tree/master/mod_statsdx maybe setting its option `hooks` to `true`

**Reply by rhariady** [reply]
thanks for the reply @badlop. I will check on these modules
